### PR TITLE
Renaming and cleaning up search services

### PR DESCRIPTION
**Pull request by MikeAlhayek**

### CODE REVIEW

**Review by hishamco — APPROVED** [review]

File: src/OrchardCore/OrchardCore.Search.AzureAI.Core/Models/AzureAISearchDefaultOptions.cs [hunk 1]
```diff
@@ -123,14 +123,14 @@ public class AzureAISearchDefaultOptions
     public void SetConfigurationExists(bool configurationExists)
         => _configurationExists = configurationExists;
 
-    public bool IsConfigurationExists()
+    public bool DoesConfigurationExist()
```
Comment (Skrypt) [inline src/OrchardCore/OrchardCore.Search.AzureAI.Core/Models/AzureAISearchDefaultOptions.cs:126]: Naming convention? I'm not sure we should use "Does". Just an opinion.

Comment (hishamco) [reply src/OrchardCore/OrchardCore.Search.AzureAI.Core/Models/AzureAISearchDefaultOptions.cs:126]: Oops, I though it's `IsConfigurationExists()`

**Review by hishamco — CHANGES_REQUESTED** [review]
I totally agree wirh @Skrypt, I though the rename from `DoesConfigurationExists()` to `IsConfigurationExists()`

I prefer to keep it as its

**Comment by Skrypt** [comment]
Okay, it is proper english to use "Does" here and based on this it makes sense:

https://source.dot.net/#q=Does

Though, we never used that prefix in the past. I'm not blocking the PR.

**Comment by hishamco** [comment]
I think there's no value for this PR, the old name is fine IMHO

**Comment by Skrypt** [comment]
Yeah, the value is the fixes in the Logger strings.

**Comment by hishamco** [comment]
Agree

**Comment by MikeAlhayek** [comment]
`IsConfigurationExists` is broken English. @sebastienros mentioned the same thing on the meeting today. 

Instead of using `Does` I changed to `HasConfiguration()` and `HasFileConfigurations()`

**Comment by hishamco** [comment]
We could use `ConfigurationExists()` such as https://learn.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.testtools.uitest.extension.searchconfiguration.configurationexists?view=visualstudiosdk-2019

**Comment by MikeAlhayek** [comment]
yes we could. But I think `Has prefix is fine too.

**Review by Skrypt — APPROVED** [review]

**Review by hishamco — APPROVED** [review]

**Pull request merged**